src/Swarm/Game/Value.hs

```diff
@@ -43,6 +46,8 @@ data Value where
   VDir :: Direction -> Value
   -- | A boolean.
   VBool :: Bool -> Value
+  -- | An injection into a sum type.  False = left, True = right.
+  VInj :: Bool -> Value -> Value
```

Comment:
Any reason you went with that representation instead of `VInl` and `VInlr`?

Reply:
Merely to cut down on the number of pattern matches needed later, e.g. when reducing a `case` there's just a single code path which uses the `Bool` to select between the continuations.  It really doesn't make that much of a difference, and I could be convinced it would be cleaner/more transparent to use separate `VInl` and `VInr`.

Reply:
I tried changing it but then I realized there's an even better reason to do it this way: implementing comparison (`inl _` is always less than `inr _`, and when the tags match the underlying values are compared) is *much* nicer when we use `VInj`: we can just `compare` the bools, and then `compare` the values if they are equal.  With `VInl` and `VInr` we need to tediously list out the four possible cases.